src/Components.sol

```diff
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.6.2 <0.9.0;
+
+import "./console.sol";
```

Comment:
_(arbitrary comment location to start a thread)_

I think we might need to fix some input ordering, or use explicit import syntax instead? Just dropped this branch into a repo that works with the current version and got a lot of errors
<img width="743" alt="image" src="https://user-images.githubusercontent.com/17163988/192615231-c4f232a9-bde3-4fe3-ae36-dcbd915ae886.png">

Reply:
Seems the issue is that I have scripts that are `is Script`, tests that are `is Test`, and those tests import and run the script which results in duplicate declarations. For example:

```solidity
// script/MyScript.s.sol
import "forge-std/Script.sol";
contract MyScript is Script {
  function run() public {
    // --- snip ---
  }
}

// test/MyTest.t.sol
import "forge-std/Test.sol";
import "script/MyScript.s.sol";

contract MyTest is MyScript, Test {
  function setUp() public {
    // Deploys the system and saves the deploy addrs to storage
    MyScript.run();
  }
}
```

Reply:
Is this a common pattern? We can fix this with a common abstract contract, for `TestBase` and `ScriptBase`, but `vm` cannot be `Vm` and `VmSafe` at the same time.

Reply:
Thanks for the insight.

What do you think about `vms`? Maybe even as an alias.

Reply:
I could be convinced, but initial reaction is that `vm` and `vms` are very similar so it's too easy to mix one up for the other without noticing

Reply:
Yeah. `vmSafe` is good for now.

You can add a new file with a common abstract contract locally and inherit it in `TestBase` and `ScriptBase` if you want to continue testing (I'm not on my computer, or you can push the fix if you want).